buildSrc/src/main/kotlin/KotlinCommunity.kt

```diff
@@ -37,4 +36,34 @@ fun addDevRepositoryIfEnabled(repositoryHandler: RepositoryHandler, project: Pro
     repositoryHandler.maven {
         url = URI.create(devRepoUrl)
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Should be used for running against non-released Kotlin compiler on a system test level.
+ *
+ * @return a Kotlin API version parametrized from command line or gradle.properties, null otherwise
+ */
+fun getOverriddenKotlinApiVersion(project: Project): String? {
+    val apiVersion = project.rootProject.properties["kotlin_api_version"] as? String
+    return if (!apiVersion.isNullOrBlank()) {
+        project.logger.info("""Configured Kotlin API version: '$apiVersion' for project $${project.name}""")
+        apiVersion
+    } else {
+        null
+    }
+}
+
+/**
+ * Should be used for running against non-released Kotlin compiler on a system test level.
+ *
+ * @return a Kotlin Language version parametrized from command line or gradle.properties, null otherwise
+ */
+fun getOverriddenKotlinLanguageVersion(project: Project): String? {
+    val languageVersion = project.rootProject.properties["kotlin_language_version"] as? String
+    return if (!languageVersion.isNullOrBlank()) {
+        project.logger.info("""Configured Kotlin Language version: '$languageVersion' for project ${project.name}""")
+        languageVersion
+    } else {
+        null
+    }
+}
```

Comment:
```suggestion
fun getOverriddenKotlinApiVersion(project: Project): String? {
    val apiVersion = project.providers.gradleProperty("kotlin_api_version").orNull
    return if (!apiVersion.isNullOrBlank()) {
        project.logger.info("""Configured Kotlin API version: '$apiVersion' for project $${project.name}""")
        apiVersion
    } else {
        null
    }
}

/**
 * Should be used for running against non-released Kotlin compiler on a system test level.
 *
 * @return a Kotlin Language version parametrized from command line or gradle.properties, null otherwise
 */
fun getOverriddenKotlinLanguageVersion(project: Project): String? {
    val languageVersion = project.providers.gradleProperty("kotlin_language_version").orNull
    return if (!languageVersion.isNullOrBlank()) {
        project.logger.info("""Configured Kotlin Language version: '$languageVersion' for project ${project.name}""")
        languageVersion
    } else {
        null
    }
}
```

I recommend avoiding using the untyped Project properties and instead using the Provider API.